File: src/main/java/org/dbshell/commands/connections/ContextValueProvider.java
package org.dbshell.commands.connections;

import org.apache.commons.io.FilenameUtils;
import org.bradfordmiller.simplejndiutils.JNDIUtils;
import org.springframework.core.MethodParameter;
import org.springframework.shell.CompletionContext;
import org.springframework.shell.CompletionProposal;
import org.springframework.shell.standard.ValueProviderSupport;
import org.springframework.stereotype.Component;

import javax.naming.NamingException;
import java.io.IOException;
import java.util.List;
import java.util.stream.Collectors;
import java.io.File;

@Component
class ContextValueProvider extends ValueProviderSupport {

    @Override
    public List<CompletionProposal> complete(MethodParameter parameter, CompletionContext completionContext, String[] hints) {
        String currentInput = completionContext.currentWordUpToCursor();
        List<CompletionProposal> proposals = null;
        try {
            proposals =
                JNDIUtils.getAvailableJndiContexts(null)
                 .stream()
                 .filter(c -> c.contains(currentInput))
                 .map(p -> FilenameUtils.removeExtension(new File(p).getName()))
                 .map(CompletionProposal::new)
                 .collect(Collectors.toList());

        } catch (NamingException e) {
            e.printStackTrace();
        } catch (IOException e) {
            e.printStackTrace();
        }
        return proposals;
    }
}

File: src/main/kotlin/org/dbshell/providers/ContextValueProvider.kt
package org.dbshell.providers

import org.dbshell.connections.ConnectionRepository
import org.springframework.shell.CompletionContext
import org.springframework.shell.CompletionProposal
import org.springframework.shell.standard.ValueProvider
import org.springframework.stereotype.Component

@Component
class ConnectionValueProvider: ValueProvider {

    override fun complete(
        completionContext: CompletionContext
    ): MutableList<CompletionProposal> {

        val currentInput = completionContext.currentWordUpToCursor()

        val proposals: MutableList<CompletionProposal> =

            ConnectionRepository.loadAllConnections()
            .filter{c -> c.key.contains(currentInput!!)}
            .map{cp -> CompletionProposal(cp.key)}
            .toMutableList()

        return proposals
    }
}